src/common/copier_config/copier_config.cpp

```diff
@@ -39,5 +39,21 @@ std::string CopyDescription::getFileTypeName(FileType fileType) {
     }
 }
 
+CopyDescription::FileType CopyDescription::getFileType(std::string& fileName) {
+    if (fileName.ends_with("parquet"))
+        return FileType::PARQUET;
+    if (fileName.ends_with("npy"))
+        return FileType::NPY;
+    if (fileName.ends_with("csv"))
+        return FileType::CSV;
+    // for "filename.csv.gz"
```

Comment:
This does not seem to be a very robust way of checking `.csv.gz`. It seems to match all of `.csv.xxx`. I think we should explicitly check for extension.